javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/MethodReferenceExprContext.java

```diff
@@ -198,7 +189,40 @@ private List<ResolvedType> inferArgumentTypes() {
         throw new UnsupportedOperationException();
     }
 
-    private int pos(MethodCallExpr callExpr, Expression param) {
+    private List<ResolvedType> resolveLambdaTypes(ResolvedType lambdaType) {
+        // Get the functional method in order for us to resolve it's type arguments properly
+        Optional<MethodUsage> functionalMethodOpt = FunctionalInterfaceLogic.getFunctionalMethod(lambdaType);
+        if (functionalMethodOpt.isPresent()) {
+            MethodUsage functionalMethod = functionalMethodOpt.get();
+
+            List<ResolvedType> resolvedTypes = new ArrayList<>();
+
+            for (ResolvedType type : functionalMethod.getParamTypes()) {
+                InferenceContext inferenceContext = new InferenceContext(typeSolver);
+
+                // Resolve each type variable of the lambda, and use this later to infer the type of each
+                // implicit parameter
+                inferenceContext.addPair(new ReferenceTypeImpl(functionalMethod.declaringType()), lambdaType);
+
+                // Now resolve the argument type using the inference context
+                ResolvedType argType = inferenceContext.resolve(inferenceContext.addSingle(type));
+
+                ResolvedLambdaConstraintType conType;
+                if (argType.isWildcard()){
+                    conType = ResolvedLambdaConstraintType.bound(argType.asWildcard().getBoundedType());
+                } else {
+                    conType = ResolvedLambdaConstraintType.bound(argType);
+                }
+
+                resolvedTypes.add(conType);
+            }
+
+            return resolvedTypes;
+        }
+        throw new UnsupportedOperationException();
+    }
+
+    private int pos(NodeWithArguments<?> callExpr, Expression param) {
```

Comment:
You can perhaps improve this method slightly by re-using NodeWithArguments.getArgumentPosition(Expression arg)